test/typescript.ts

```diff
@@ -8,7 +8,7 @@ store.dispatch(dispatch => {
   dispatch({type: 'FOO'});
 });
 
-store.dispatch((dispatch, getState) => {
+store.dispatch((dispatch, getState: () => {foo: string}) => {
```

Comment:
The type for `getState` here should not be needed.
Does it not work without it?

Reply:
In Redux v4, Dispatch doesn't take any state object as a generic type. So as far as I can tell, there's no way to get the Store's State type into the Thunk arguments.

Reply:
Oh, okay, I pulled it and now I see the problem.
Yeah, I'll stick to my overrides for now :( Thank you for your time.

My current code:

```typescript
export type AppStore = Store<AppState> & {
  dispatch<R>(asyncAction: AsyncAction<R>): R;
};

export type AsyncAction<R = any> = (dispatch: AppStore['dispatch'], getState: AppStore['getState']) => R;

// store.js
const store = createStore(…) as AppStore; // :(
```